R/lint.R

```diff
@@ -24,6 +24,10 @@ NULL
 #' character string, store the cache in this directory.
 #' @param ... additional arguments passed to \code{\link{exclude}}.
 #' @param parse_settings whether to try and parse the settings.
+#' @param text Optional argument for supplying a string or lines or lines directly,
+#'   e.g. if the file is already in memory or linting is being done ad hoc.
+#'   If \code{text} is supplied but \code{filename} is missing, then \code{tempfile()}
+#'   will be used as the \code{filename}.
```

Comment:
it looks like `<text>` will be used as the file name, this reads like the output would have the tempfile path instead... could you clarify what we mean here?

Reply:
It is indeed a bit ambiguous here. Here I mean that if `text` is provided but `filename` is not, then we'll write the text to a temp file (same behavior as before when `filename` is detected to be code), but the lint result will show as `<text>` which I believe should be the desired output rather than showing the tempfile in the lint result.

Looks like I should update this doc here, maybe just removing the `text` without `filename` case in the doc looks enough?

Reply:
`a string or lines or lines directly` looks a bit redundant?

Reply:
I think so. Removed the last sentence